File: pdv.api/app/models/sale_model.py
from pydantic import BaseModel
from models.product_model import productSaleModel
from models.paymentMethod_model import PaymentMethodModel

class saleModel(BaseModel):
    date: str = None
    client: str = None
    paymentMethod: PaymentMethodModel
    totalValue: float = None
    products: list = productSaleModel
    pending: bool = False
    address: str = None
    delivery: bool = False

File: pdv.view/src/app/models/sale.model.ts
import { PaymentMethodModel } from './PaymentMethodModel.modal';
import { productCart } from './productCart.model';
export class sale{
    date: string = '';
    client: string = '';
    totalValue: number = 0;
    paymentMethod: PaymentMethodModel = new PaymentMethodModel();
    products: Array<productCart> = [];
    pending: boolean = false;
    address: string = '';
    delivery: boolean = false;
    deliveryTime: Date = new Date();
}